### ISSUE TITLE
'args.yaml' file missed parameter(s) while using 'rtdetr-l.yaml' for training

### ISSUE THREAD

**Issue by A1exan10er** [ultralytics/ultralytics#3739]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

Hello ultralytics, 

firstly, thank you very much for updating the **RT-DETR** in available YOLOv8 models.

Sadly, I met a problem while using **rtdetr-l.yaml** for training.
Training itself has no problem, it can run successfully. But in its generated folder (e.g. 'train' folder), the file **args.yaml** probably missed 1 (or more) parameter(s).
One example is here: 
![Screenshot from 2023-07-14 13-05-30](https://github.com/ultralytics/ultralytics/assets/49768349/4925d721-16e0-4260-a8a6-249cfb94ad2a)
The third line is info for used model, and I believe it should show **_rtdetr-l.yaml_** or the **path leads to it**.
In my case, it's _**null**_, so I think this may be a mistake.
And probably because here doesn't have the correct value, in the **corresponding project run** on **wandb**, it also missed this part.

I used the same code on **docs for RT-DETR**, section **Python API** ([https://docs.ultralytics.com/models/rtdetr/?h=rt#python-api](url)).

Other than the **Python API** demo code, do I need to write something extra to assign some specific parameters as some fixed value/string? Or is it because some part responsible for recording args is not running?

Hope to hear from you soon!
Tianyu

### Environment

- YOLO: Ultralytics YOLOv8.0.134 🚀 Python-3.8.16 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24245MiB)
- OS: Ubuntu 20.04
- Python: 3.8.16

### Minimal Reproducible Example
```
from ultralytics import RTDETR
import wandb


model = RTDETR("rtdetr-l.yaml")
model.info()

train_data = 'blended_floor.yaml' # Set training data path
model.train(data=train_data)   # train


wandb.init(
    # set the wandb project where this run will be logged
    project="YOLOv8-rtdetr",
    
    # track hyperparameters and run metadata
    config={
    "architecture": "RT-DETR",
    "dataset": "dataset_2019", # 'blended_floor.yaml'
    }
)

wandb.finish()
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@A1exan10er yes you're right! but it won't affect the training, only the args.yaml looks incomplete. This is because we have a brand new class for RTDETR which is not as featured as YOLO class, and we'll fix this by introducing a BaseModel class for these model in a few days. :)
BTW you don't need to worry as the training still goes correct, it only make the args.yaml look incomplete.

**Comment by Laughing-q** [comment]
@glenn-jocher I'll work on the BaseModel class after refactor PR is all set.

**Comment by A1exan10er** [comment]
> @A1exan10er yes you're right! but it won't affect the training, only the args.yaml looks incomplete. This is because we have a brand new class for RTDETR which is not as featured as YOLO class, and we'll fix this by introducing a BaseModel class for these model in a few days. :) BTW you don't need to worry as the training still goes correct, it only make the args.yaml look incomplete.

Thank you for the quick answer! Look forward to seeing the update! 🤩

**Comment by glenn-jocher** [comment]
Hi @A1exan10er! Thank you for bringing this issue to our attention. I apologize for the incomplete args.yaml file in the rtdetr-l.yaml model. The missing information does not affect the training process, and your training will still run correctly. However, I understand the importance of having accurate and complete information in the args.yaml file, and I want to assure you that we are working on a solution.

We will be introducing a new BaseModel class specifically for models like RTDETR, which will ensure that the args.yaml file contains all the necessary information. This update will be implemented in the next few days, so please stay tuned for the release.

In the meantime, I want to reassure you that despite the incomplete args.yaml file, your training is proceeding correctly. Thank you for your understanding, and if you have any further questions or concerns, please don't hesitate to ask.

**Comment by glenn-jocher** [comment]
@A1exan10er great news 😃! Your original issue may now be resolved ✅ in PR #3934. Now when I run your code I get the model name showing in args.yaml:

```python
from ultralytics import RTDETR

model = RTDETR("rtdetr-l.yaml")

train_data = 'coco8.yaml' # Set training data path
model.train(data=train_data)   # train
```
 
<img width="874" alt="Screenshot 2023-07-30 at 22 20 17" src="https://github.com/ultralytics/ultralytics/assets/26833433/89c327a0-8c83-4aa1-af4b-02655ab39eb3">


To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
